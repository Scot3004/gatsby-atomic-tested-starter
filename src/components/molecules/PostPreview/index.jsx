import React from 'react'
import Card from '../../atoms/Card'
import CardContent from '../../atoms/CardContent'
import Avatar from '../../atoms/Avatar'
import PropTypes from 'prop-types'
import Excerpt from '../../atoms/Excerpt'
import { Title } from '../../atoms/Title'

export const PostPreview = props => (
  <Card>
    <Avatar src={props.image} alt={props.title} size='64px' />
    <CardContent>
      <Title>{props.title}</Title>
      <Excerpt>{props.excerpt}</Excerpt>
    </CardContent>
  </Card>
)

PostPreview.propTypes = {
  image: PropTypes.string,
  title: PropTypes.string,
  excerpt: PropTypes.string
}

export default PostPreview
