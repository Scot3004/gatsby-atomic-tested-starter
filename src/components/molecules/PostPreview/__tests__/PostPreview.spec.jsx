import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import PostPreview from '..';

const stub = {
  title: 'Title stub',
  excerpt: 'Excerpt stub',
  image: 'image_stub',
  slug: 'slug_stub'
};

test('it works', () => {
  const tree = renderer.create(<PostPreview {...stub} />).toJSON();
  expect(tree).toMatchSnapshot();
});
