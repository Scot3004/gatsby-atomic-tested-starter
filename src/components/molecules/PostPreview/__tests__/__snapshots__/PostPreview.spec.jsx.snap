// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c1 {
  -webkit-flex: 0 0 64px;
  -ms-flex: 0 0 64px;
  flex: 0 0 64px;
  width: 64px;
  height: 64px;
  padding: 1rem;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 0 1rem 0;
  padding: .6rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  width: 100%;
}

.c0:hover {
  box-shadow: 0 6px 12px 0 rgba(0,0,0,0.2);
}

.c0:last-child {
  margin: 0;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c4 {
  margin: 0;
}

.c3 {
  margin: 0 0 .6rem 0;
  padding: 0;
  word-wrap: none;
}

<div
  className="c0"
>
  <img
    alt="Title stub"
    className="c1"
    size="64px"
    src="image_stub"
  />
  <div
    className="c2"
  >
    <a>
      <h2
        className="c3"
      >
        Title stub
      </h2>
    </a>
    <p
      className="c4"
    >
      Excerpt stub
    </p>
  </div>
</div>
`;
